From 815459981575a9ed5f340d1ede9de7114039ea80 Mon Sep 17 00:00:00 2001
From: Vipin Chandran <vchandran@axiado.com>
Date: Sat, 3 May 2025 00:57:04 -0700
Subject: [PATCH] Ensure post-code manager starts after lpcsnoop

- Added `Requires` and `After` directives to ensure post-code manager starts after lpcsnoop.service.
- Enabled automatic restart with `Restart=always` and `RestartSec=5` for automatic recovery

Upstream-Status: Pending

---
 .../xyz.openbmc_project.State.Boot.PostCode@.service          | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/service_files/xyz.openbmc_project.State.Boot.PostCode@.service b/service_files/xyz.openbmc_project.State.Boot.PostCode@.service
index db21c21..906cebc 100644
--- a/service_files/xyz.openbmc_project.State.Boot.PostCode@.service
+++ b/service_files/xyz.openbmc_project.State.Boot.PostCode@.service
@@ -1,7 +1,11 @@
 [Unit]
 Description=Post code manager (host %i)
+Requires=lpcsnoop.service
+After=lpcsnoop.service
 
 [Service]
+Restart=always
+RestartSec=5
 ExecStart=/usr/bin/env post-code-manager --host %i
 SyslogIdentifier=post-code-manager%i
 Type=dbus
-- 
2.25.1

